### PR TITLE
SPARKC-331: Include More Index Information in TableDef

### PR DESCRIPTION
**Pull request by RussellSpitzer**
In this patch we add the IndexClassName to the TableDef inorder to use
this information in pushdowns. All though the current C* driver does not
expose indexOptions information at this time it will be needed for
dealing with SASI indexes in the future.

### CODE REVIEW

File: spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/cql/Schema.scala [hunk 1]
```diff
@@ -87,12 +87,15 @@ case class ClusteringColumn(index: Int) extends ColumnRole
...
+/** A Cassandra column metadata that can be serialized.  */
 case class ColumnDef(
-    columnName: String,
-    columnRole: ColumnRole,
-    columnType: ColumnType[_],
-    indexed : Boolean = false) extends FieldDef {
+  columnName: String,
+  columnRole: ColumnRole,
+  columnType: ColumnType[_],
+  indexClassName: Option[String] = None,
```
Comment (jacek-lewandowski) [inline spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/cql/Schema.scala]: I would really create a case class `IndexDef`

**Comment by RussellSpitzer** [comment]
@jacek-lewandowski updated, this is a bit of a different implementation but should be most flexible going forwards :)

File: spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/cql/Schema.scala [hunk 2]
```diff
@@ -128,12 +136,35 @@ case class TableDef(
...
   require(clusteringColumns.forall(_.isClusteringColumn), "All clustering columns must have role ClusteringColumn")
   require(regularColumns.forall(!_.isPrimaryKeyColumn), "Regular columns cannot have role PrimaryKeyColumn")
-  
+
+  val allColumns = regularColumns ++ clusteringColumns ++ partitionKey
+
+  val indexesForColumnName: Map[String, Seq[IndexDef]] = indexes.groupBy(_.target)
+
+  val indexesForColumnDef: Map[ColumnDef, Seq[IndexDef]] = {
+    val nameToColumnDef: Map[String, ColumnDef] = allColumns
```
Comment (jacek-lewandowski) [inline spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/cql/Schema.scala]: we already have `columnByName` in `TableDef`

**Comment by jacek-lewandowski** [comment]
+1

File: spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/cql/Schema.scala (hunk 2 above)
Comment (RussellSpitzer) [inline spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/cql/Schema.scala]: good catch

**Pull request merged**